pkg/network/dhcpconfigurator/dhcp_configurator.go

```diff
@@ -0,0 +1,79 @@
+package dhcpconfigurator
```

Comment:
As far as I'm aware, [this is](https://github.com/kubevirt/kubevirt/blob/b7a0b1164be47c34e572977bcc2830d297ba23cf/pkg/virt-launcher/virtwrap/network/dhcp/dhcp.go#L50) the dhcp server.

I'd prefer to have a 1 to 1 relationship between names and things instead of 3 things called dhcp server.

Reply:
As I see it, [this is the generic server](https://github.com/kubevirt/kubevirt/blob/b7a0b1164be47c34e572977bcc2830d297ba23cf/pkg/virt-launcher/virtwrap/network/dhcp/dhcp.go#L50) and the stuff we have on this PR is the VMI server

So what do you think about
`dhcp/vmiserver.go`

And change all the stuff according, for me `DhcpConfiguartor` means a little and that something that calls `Configurator` has a function called `EnsrueDhcp` or `startServer` does not make much sense

Reply:
I had hoped the reader would read dhcp configurator and conclude "this configures dhcp". 

You have to understand that what you call "dhcp server" does more than serve; it exports the dhcp configuration from one process that has the pod interface info to another that doesn't. I want to have an entity that encapsulates all related dhcp operations, and is used by any binding that requires dhcp. 

Regarding your `VMIServer` proposal, I think that's redundant in a virtualization project. My point is that its fair to assume the dhcp server is serving the vm.

Reply:
Let's go with Configurator but let's not be redundant with package name, we can have `dhcp/configurator.go` and then `struct Configurator` in the future we can move the server here so we will have `dhcp/server.go` and `struct Server` so it will be qualified as `dhcp.Configurator`and `dhcp.Server` and the constructors will be `dhcp.NewConfigurator` and `dhcp.NewServer`.